doc/mgr/rgw.rst

```diff
@@ -18,57 +19,120 @@ RGW Realm Operations
 
 Bootstrapping RGW realm creates a new RGW realm entity, a new zonegroup,
 and a new zone. It configures a new system user that can be used for
-multisite sync operations, and returns a corresponding token. It sets
-up new RGW instances via the orchestrator.
+multisite sync operations. Under the hood this module instructs the
+orchestrator to create and deploy the corresponding RGW daemons. The module
+supports both passing the arguments through the cmd line or as a spec file:
 
-It is also possible to create a new zone that connects to the master
-zone and synchronizes data to/from it.
+.. prompt:: bash #
+
+  rgw realm bootstrap [--realm-name] [--zonegroup-name] [--zone-name] [--port] [--placement] [--start-radosgw]
+
+The command supports providing the configuration through a spec file (`-i option`):
+
+.. prompt:: bash #
+
+  ceph rgw realm bootstrap -i myrgw.yaml
+
+Following is an example of RGW mutlisite spec file:
+
+.. code-block:: yaml
+
+  rgw_realm: myrealm
+  rgw_zonegroup: myzonegroup
+  rgw_zone: myzone
+  placement:
+    hosts:
+     - ceph-node-1
+     - ceph-node-2
+  spec:
+    rgw_frontend_port: 5500
+
+.. note:: The spec file used by RGW has the same format as the one used by the orchestrator. Thus,
+          the user can provide any orchestration supported rgw parameters including advanced
+          configuration features such as SSL certificates etc.
+
+Users can also specify custom zone endpoints in the spec (or through the cmd line). In this case, no
+cephadm daemons will be launched. Following is an example RGW spec file with zone endpoints:
```

Comment:
> to clarify, this is saying if certain fields are added in the spec, it won't pass the spec to the orchestrator for daemon deployment?

The answer is yes, because in case the user specifies custom endpoints we cannot force cephadm to use them and IMHO it doesn't make sense to launch cephadm daemons in this case because we won't be able to listen in the specified endpoints. WDYT?

Reply:
eh, I think it's a bit of a confusing system that some certain field being present in the spec decides whether we'll tell the orchestrator to deploy daemons, but I'm okay with it for now and maybe we can come up with some more explicit way of handling it later.

Reply:
Yeah, I agree. Probably we have to show some more explicit warning message in this case as rgw daemons are not launched by cephadm. I  expect this something that a user would use only if they are deploying the daemons separately themselves. Anyway this is something that we could address later :+1: